bookstore/client/nb_client.py

```diff
@@ -30,6 +30,7 @@
 
 
 def extract_kernel_id(connection_file):
+    # TODO should this revert to not using regex ?
```

Comment:
Yes.

Reply:
No I disagree. The previous version was riddled with bugs. Regex is the only clean way to extract a prefix in python. `connection_filename.lstrip("kernel-") `  would strip all instances of the characters "e" "k" "r" "n" "l" and "-". So, when the connection file began with an "e" this would result in being unable to find the connection file.